### ISSUE TITLE
Handle the post meta better for all post types

### ISSUE THREAD

**Issue by joyously** [Automattic/_s#1203]
This function has a bunch of code that is unnecessary and does not handle all cases.

https://github.com/Automattic/_s/blob/67cdccbb674baf25ccb9538626af6dbf043c2e1c/inc/template-tags.php#L48

```PHP
// Hide category and tag text for pages.
if ( 'post' === get_post_type() ) {
	/* translators: used between list items, there is a space after the comma */
	$categories_list = get_the_category_list( esc_html__( ', ', '_s' ) );
	if ( $categories_list ) {
		/* translators: 1: list of categories. */
		printf( '<span class="cat-links">' . esc_html__( 'Posted in %1$s', '_s' ) . '</span>', $categories_list ); // WPCS: XSS OK.
	}
	/* translators: used between list items, there is a space after the comma */
	$tags_list = get_the_tag_list( '', esc_html_x( ', ', 'list item separator', '_s' ) );
	if ( $tags_list ) {
		/* translators: 1: list of tags. */
		printf( '<span class="tags-links">' . esc_html__( 'Tagged %1$s', '_s' ) . '</span>', $tags_list ); // WPCS: XSS OK.
	}
}
```

Instead, it could be 
`the_taxonomies();`
which would handle posts, pages, and custom post types. Of course, the arguments passed can customize the output.

**Comment by grappler** [comment]
Depending on the how the custom taxonomies are used I don't think they should be displayed by default.

Now that `_s` supports WooCommerce it would be interesting to look at how the product categories should be displayed as a concrete example.

**Comment by samikeijonen** [comment]
Justin have good [hybrid_get_post_terms()](https://github.com/justintadlock/hybrid-core/blob/b55b640682ec9c97ccee576360c12df0fa2a6e6b/inc/template-post.php#L122) function for this if it's ever needed in `_s`.

**Comment by crunnells** [comment]
I agree with @grappler's comment above, and I'm sure there are many other situations where we don't want to display post meta on all post types. I'd say we can leave this up to the theme author to modify as they see fit.
